src/index.js

```diff
@@ -168,6 +172,9 @@ class Libp2p extends EventEmitter {
     this.peerRouting = peerRouting(this)
     this.contentRouting = contentRouting(this)
 
+    // Keychain
+    this.keychain = this._options._keychain || new NoKeychain()
```

Comment:
Why `NoKeychain`? It's just throwing errors whenever you try to use it. Just leave it undefined and let type errors happen.